### PR TITLE
Make mNumFreeBuffers unsigned

### PR DESCRIPTION
**Pull request by aeliot**
Make mNumFreeBuffers a uint rather then an int since the number of free buffers should never be negative.

### CODE REVIEW

File: src/core/common/message.cpp [hunk 1]
```diff
@@ -148,7 +148,7 @@ ThreadError MessagePool::FreeBuffers(Buffer *aBuffer)
 
 ThreadError MessagePool::ReclaimBuffers(int aNumBuffers)
 {
-    int numFreeBuffers;
+    uint16_t numFreeBuffers;
```
Comment (pvanhorn) [inline src/core/common/message.cpp]: With this change aNumBuffers will be promoted to unsigned in the comparison on line 159, I think?  You are welcome to prove me wrong.

Comment (aeliot) [inline src/core/common/message.cpp]: Yea, I just tested it and it looks like your right..... I honestly didn't even know C would do that....

Comment (aeliot) [inline src/core/common/message.cpp]: I added a check for negative values that should get around this issue.

**Comment by pvanhorn** [comment]
LGTM

File: include/platform/messagepool.h [hunk 2]
```diff
@@ -69,7 +69,7 @@ extern "C" {
  * @param[in] aBufferSize          The size in bytes of a Buffer object.
  *
  */
-void otPlatMessagePoolInit(int aMinNumFreeBuffers, size_t aBufferSize);
+void otPlatMessagePoolInit(uint16_t aMinNumFreeBuffers, size_t aBufferSize);
```
Comment (librasungirl) [inline include/platform/messagepool.h]: also update type (`int` -> `uint16_t`) in the introduction of `param[in]` in Line 68 ?

Comment (jwhui) [inline include/platform/messagepool.h]: @aeliot, what do you think on updating the comment?

Comment (aeliot) [inline include/platform/messagepool.h]: Oh, sorry, I completely missed the notification for this.  I'll make that change now.

**Pull request merged**